### ISSUE TITLE
Accessibility - fix keyboard focus for popups/dialogs (e.g. support), change secondary form button (e.g. cancel) style

### ISSUE THREAD

**Issue by kaitlinnewson** [IQSS/dataverse#5717]
Tested in 4.10 and 4.11 in both Scholars Portal and Harvard Dataverses, in Firefox and Chrome (OSX).

When opening an overlay, e.g. the Support form, the user's keyboard focus should go there. The overlay is automatically added to the end of the tab order, meaning the user has to tab through the whole page underneath before they can get to the overlay window.

**Comment by TaniaSchlatter** [comment]
Hi @Kaitlinnewson, thanks for opening this issue. We are working on improving accessibility when working on related functionality.  Its helpful to raise the visibility of this and other accessibility issues. Have you come across any best practices for addressing accessible popups?

**Comment by kaitlinnewson** [comment]
Hi @TaniaSchlatter, this article from uxplanet outlines some best practices (both for ux and accessibility): https://uxplanet.org/best-practices-for-modals-overlays-dialog-windows-c00c66cddd8c

**Comment by TaniaSchlatter** [comment]
I also found an accessibility plugin for Bootstrap that addresses popup focus. I am not sure if this might be a fit, but the team will review these references and make a plan. https://github.com/paypal/bootstrap-accessibility-plugin.

**Comment by mheppler** [comment]
Was able to add a PrimeFaces `p:focus` component inside the `h:form` in the `p:dialog` for the Support popup, and added render logic to the `for` attribute.

![Screen Shot 2019-05-21 at 5 05 56 PM](https://user-images.githubusercontent.com/687227/58130902-3e0bc300-7beb-11e9-940e-5e6150320464.png)

Will need to review and fix the other 95 dialogs, across 23 files...

- [x] contactFormFragment.xhtml, "contactDialog"
- [ ] dashboard-datamove.xhtml, "datamoveConfirmation"
- [ ] dashboard-users.xhtml, "toggleSuperuserConfirmation", "removeRolesConfirm"
- [ ] dataset-license-terms.xhtml, "previewGuestbook"
- [ ] dataset-widgets.xhtml, (no id for "removeThumbnail"), "selectFileThumbnail"
- [ ] dataset.xhtml (38 POPUPS!!!), "shareDialog", (no id for "downloadInvalid"), (no id for "downloadMixed"), (no id for "deleteConfirmation"), (no id for "deleteVersionConfirmation"), "privateUrlId", (no id for "disablePrivateUrlConfirmation"), (no id for "deleteSelectedFileConfirmation"), (no id for "computeInvalid"), "deaccessionBlock", "deaccessionConfirmation", "deaccessionAllConfirmation", "compareTwo", "detailsBlocks", "selectFilesForDownload", "selectFilesForRequestAccess", "selectFilesForDelete", "selectFilesForRestrict", "selectFilesForUnRestrict", "selectFilesForEditMetadata", "selectFilesForEditTags", "fileTagsPopup", (no id for "accessSignUpLogIn_popup"), (no id for "configureToolPopup"), "downloadPopup", "downloadPackagePopup", "requestAccessPopup", "downloadDataSubsetPopup", "linkDatasetForm", "computeBatchListPopup", "accessPopup", "inreview", (no id for "publishConfirm"), (no id for "publishParent"), (no id for "releaseDraft"), (no id for "mayNotRelease"), (no id for "maynotPublishParent"), (no id for "sendBackToContributor")
- [ ] dataverse.xhtml, "shareDialog", "browseOptsForm", (no id for "confirmation"), (no id for "mayNotRelease"), (no id for "deleteConfirmation"), "linkDataverseForm", "resetModifications"
- [ ] dataverse_template.xhtml, "statusInfo"
- [ ] editFilesFragment.xhtml, "fileTypeDifferentPopup", "selectFilesForDeleteFragment", (no id for "deleteFileConfirmation"), "editFileAccessPopup", "fileAdvancedOptions", "fileSetThumbnail", "fileThumbnailConfirm", "editFileTagsPopup"
- [ ] explicitGroup-new-dialog.xhtml, "explicitGroupNewDialog"
- [ ] file-configure-dropdown-fragment.xhtml, (no id for "mapData_popup")
- [ ] file.xhtml, "shareDialog", (no id for "deleteFileConfirmation"), "downloadPopup", "downloadDataSubsetPopup", "downloadPackagePopup", (no id for "configureToolPopup"), (no id for "accessSignUpLogIn_popup", "accessPopup", "fileAlreadyReplacedDraft", "fileAlreadyReplacedPrevious", (no id for "computeAccessDeniedPopup")
- [ ] harvestclients.xhtml, "deleteConfirmation", "newHarvestingClientDialog" 
- [ ] harvestsets.xhtml, "deleteConfirmation", "newOAISetDialog"
- [ ] manage-groups.xhtml, (no id for "deleteConfirmation"), "viewGroup"
- [ ] manage-guestbooks.xhtml, (no id for "deleteConfirmation"), "previewGuestbook"
- [ ] manage-templates.xhtml, (no id for "deleteConfirmation"), "viewTemplate"
- [ ] permissions-manage-files.xhtml, "viewRemoveDialog", "assignDialog", (no id for "confirmation")
- [ ] permissions-manage.xhtml, "accessDialog", (no id for "confirmation")
- [ ] provenance-popups-fragment.xhtml, "editProvenancePopup", "confirmProvenancePopup"
- [ ] roles-assign.xhtml, "userGroupDialog", "confirmFileDownloader"
- [ ] roles-edit.xhtml, "rolesPermissionsDialog"
- [ ] search-include-fragment.xhtml, (no id for "addData_popup")

**Comment by TaniaSchlatter** [comment]
We are increasing the scope of this issue:
-  to add keyboard focus form pages
-  to add a secondary button class (link text) to form pages, dialog boxes, and confirmation popups.

**Comment by pdurbin** [comment]
I just scrolled through the thousands of lines of code changes in pull request #5903 as of fd05813 and nothing jump out at me as being wrong or bad (the addition of `webmanifest` to `web.xml` gave me pause but the change is explained in the pull request) but this is obviously a lot of code and we will have to rely on testing to confirm that nothing is broken.

Toward that end @mheppler has provided a long and extremely detailed list of what to test at https://github.com/IQSS/dataverse/pull/5903#issuecomment-498762156

When I say long and detailed, I really mean it. Here's a thumbnail of how long the list is:

![Screen Shot 2019-06-11 at 12 04 11 PM](https://user-images.githubusercontent.com/21006/59288182-28654880-8c41-11e9-8585-96e10a99224d.png)

This strikes me as more than "Medium".

I went ahead and spun up the branch to get a sense of how the buttons look. Here are a couple examples. Note how "Cancel" is just a text link:

![Screen Shot 2019-06-11 at 11 57 59 AM](https://user-images.githubusercontent.com/21006/59288244-58145080-8c41-11e9-9938-47d7a1b05a66.png)

![Screen Shot 2019-06-11 at 11 58 16 AM](https://user-images.githubusercontent.com/21006/59288243-58145080-8c41-11e9-9476-295219a621dc.png)

I think having the primary button be emphasized like this will increase the usability of Dataverse so I'm excited about this change! Developers, please note that this change has also been noted in the Style Guide in 62642ef.

**Comment by kcondon** [comment]
Issues so far:
[ ] Edit Tags pop up has old cancel btn style.

**Comment by mheppler** [comment]
Reverted my passwordreset changes to resolve the issue @kcondon reported above. There is a lot going on with that page that needs to be address, and I have found a recently opened issue (#5462) that is the perfect place to park all these related UI/UX issues I discovered.

Found that the dataset versions tab issue reported above was due to the test dataset on dataverse-internal containing a previous version that was deaccessioned, which resulted in the View Differences btn and checkboxes not to be rendered.

Reported all of this in my really, really, really long comment in the PR, as well as in Slack, directly to Kevin. Should be good to re-QA, confirm and ship ⛵️.
